src/SIM/SIMbase.C

```diff
@@ -1328,9 +1328,13 @@ bool SIMbase::initSystem (int mType, size_t nMats, size_t nVec, bool withRF)
     mType = SystemMatrix::DENSE;
   }
 
+  LinAlg::LinearSystemType type = LinAlg::GENERAL_MATRIX;
+  if (myProblem)
+    type = myProblem->getLinearSystemType();
+
   return myEqSys->init(static_cast<SystemMatrix::Type>(mType),
                        mySolParams, nMats, nVec, withRF,
-                       myProblem->getLinearSystemType(), opt.num_threads_SLU);
+                       type, opt.num_threads_SLU);
 }
```

Comment:
Litt av same greie egentlig: linear-system type har vel egentlig ikke noe med integranden å gjøre? Burde kanskje heller være et flagg (member) i SIMbase scope som så kan settes av de respektive app-SIM'ene som trenger det, istedet for av integrand-objektet.